cassandra/registry.py

```diff
@@ -12,17 +12,87 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from functools import partial
 from collections import defaultdict
 
 from cassandra import ProtocolVersion
 from cassandra.protocol import *
 
 
+class ProtocolVersionRegistry(object):
+    """Default implementation of the ProtocolVersionRegistry"""
+
+    # default versions and support definition
+    protocol_version = ProtocolVersion
```

Comment:
The ProtocolVersionRegistry keeps a trace of the ProtocolVersion support class for the driver purpose. In DSE, this will be set to DseProtocolVersion which additional versions and support checks. I don't this  the `ProtocolVersion` naming is accurate but kept it only because we had that before.

Reply:
oh that was just a personal thought. I just feel a bit confusing that we use `protocol_version` everywhere in the code, which is always an integer... but that this object is called a `ProtocolVersion`... but it's only some support checks. Anyway.. it's not something I'm going to touch.